internal/aez/ccd_data.go

```diff
@@ -0,0 +1,169 @@
+package aez
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/innosat-mats/rac-extract-payload/internal/ccsds"
+)
+
+// WDWMode describes the CCD WDW parameter
+type WDWMode uint8
+
+// WDWModeManual = 0, check WDWOV for overflows
+//
+// WDWModeAutomatic, window selected depending on input data
+const (
+	WDWModeManual    WDWMode = 0
+	WDWModeAutomatic WDWMode = 1
+)
+
+// Wdw is the composite status of Image Window Mode
+type Wdw uint8
+
+// Mode returns the WDWMode type used encoded in Bit[7]
+func (wdw Wdw) Mode() WDWMode {
+	if (wdw & 0b0000001) == 0 {
+		return WDWModeManual
+	}
+	return WDWModeAutomatic
+}

```

Comment:
You have masked out bit 2, why not do the simpler
```suggestion
// Mode returns the WDWMode type used encoded in Bit[7]
func (wdw Wdw) Mode() WDWMode {
	return Wdw(wdw >> 7)
}
```

Reply:
For someone not too used with bit operations like me, a mask is immensely easier and clearer than bit-shifts. There's something intrinsic in them that hurts my brain. But as you kind of indirectly point out, is my mask really validating the 7th bit or the first?

Reply:
You're masking out bit 0 the LSB.

An illustrated example on shift left (<<) https://www.computerhope.com/jargon/b/bit-shift.htm

Reply:
It's not really helping, it would help if we flipped the convention and wrote bit numbers left to right.